solr/core/src/java/org/apache/solr/highlight/SolrHighlighter.java

```diff
@@ -88,32 +88,28 @@ protected boolean emptyArray(String[] arr) {
   }
 
   private static String[] expandWildcardsInFields(
-      Supplier<Collection<String>> availableFieldNamesSupplier, String... fields) {
+      Supplier<Collection<String>> availableFieldNamesSupplier, String... inFields) {
     Set<String> expandedFields = new LinkedHashSet<String>();
-    for (String field : fields) {
-      expandWildcardsInFields(
-          expandedFields, availableFieldNamesSupplier, SolrPluginUtils.split(field));
-    }
-    return expandedFields.toArray(new String[] {});
-  }
-
-  private static void expandWildcardsInFields(
-      Set<String> expandedFields,
-      Supplier<Collection<String>> availableFieldNamesSupplier,
-      String... fields) {
-    for (String field : fields) {
-      if (field.contains("*")) {
-        // create a Java regular expression from the wildcard string
-        String fieldRegex = field.replaceAll("\\*", ".*");
-        for (String storedFieldName : availableFieldNamesSupplier.get()) {
-          if (storedFieldName.matches(fieldRegex)) {
-            expandedFields.add(storedFieldName);
+    Collection<String> availableFieldNames = null;
+    for (String inField : inFields) {
+      for (String field : SolrPluginUtils.split(inField)) {
+        if (field.contains("*")) {
+          // create a Java regular expression from the wildcard string
+          String fieldRegex = field.replaceAll("\\*", ".*");
```

Comment:
I realize it was like this before but it'd be awesome if you could extract a Pattern here to be re-used across looping the fields instead of letting Java's convenient API methods internally recompute the automata internally each time.

Reply:
Good idea. Done in https://github.com/apache/solr/pull/767/commits/87570fccb1adeff5b7fafdb80604d9141f1e930a commit.

Reply:
Nice; thanks!

Reply:
> I realize it was like this before but it'd be awesome if you could extract a Pattern here to be re-used across looping the fields instead of letting Java's convenient API methods internally recompute the automata internally each time.

#778 explores forbidding `String.matches(String)` in the code base, in favour of compiled patterns.